sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -991,16 +996,16 @@ namespace Azure { namespace Storage { namespace DataLake {
       }
     }
 
-    static FileSystemListPathsResponse FileSystemListPaths(std::string url, std::shared_ptr<Azure::Core::Http::HttpTransport> transport, Azure::Core::Context& context, const FileSystemListPathsOptions& fileSystemListPathsOptions)
+    static FileSystemListPathsResponse FileSystemListPaths(std::string url, Azure::Core::Http::HttpPipeline& pipeline, Azure::Core::Context context, const FileSystemListPathsOptions& fileSystemListPathsOptions)
```

Comment:
Shouldn't we have a class for each resource `class Service`, `class FileSystem`, `class Path`? These functions should be member function without the resource name prefix, e.g. `FileSystem::ListPaths`

Reply:
This is a valid comment. I have adjusted the structure accordingly.